crypto/s2n_hash.c

```diff
@@ -548,7 +548,7 @@ int s2n_hash_init(struct s2n_hash_state *state, s2n_hash_algorithm alg)
 int s2n_hash_update(struct s2n_hash_state *state, const void *data, uint32_t size)
 {
     PRECONDITION_POSIX(s2n_hash_state_validate(state));
-    ENSURE_POSIX(S2N_MEM_IS_READABLE(data, size), S2N_ERR_PRECONDITION_VIOLATION);
+    ENSURE_POSIX(size == 0 || S2N_MEM_IS_READABLE(data, size), S2N_ERR_PRECONDITION_VIOLATION);
```

Comment:
Why is this new condition added? The macro is defined as
```
#define S2N_MEM_IS_READABLE(base, len) (((len) == 0) || (base) != NULL)
```
which already includes this new one, right?

Reply:
True! I removed it.